src/System.Text.Primitives/System/Text/TextEncoding.cs

```diff
@@ -121,6 +123,156 @@ public override bool TryEncodeFromUtf16(ReadOnlySpan<char> utf16, Span<byte> buf
             return true;
         }
 
+        public override bool TryEncodeFromUnicode(ReadOnlySpan<UnicodeCodePoint> codePoints, Span<byte> buffer, out int bytesWritten)
+        {
+            var avaliableBytes = buffer.Length;
+            int bytesWrittenForCodePoint = 0;
+            bytesWritten = 0;
+
+            for (int i = 0; i < codePoints.Length; i++)
+            {
+                if (!TryEncodeFromUnicode(codePoints[i], buffer.Slice(bytesWritten), out bytesWrittenForCodePoint))
```

Comment:
I don't think we can slice for every single code point. It's too expensive. I am speaking about the routines that we need to get very fast (e.g. encoding between UTF8 and UTF16 and vice versa). Some other routines (for simplicity sake) can of course slice. I am not sure if this one is one of these or not.